packages/web/src/features/languages/ImportLanguageGlossesView.tsx

```diff
@@ -0,0 +1,137 @@
+import {
+  GetLanguageImportOptionsResponseBody,
+  GetLanguageResponseBody,
+  SystemRole,
+} from '@translation/api-types';
+import { useRef, useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { Trans, useTranslation } from 'react-i18next';
+import {
+  LoaderFunctionArgs,
+  useLoaderData,
+  useNavigate,
+} from 'react-router-dom';
+import apiClient from '../../shared/apiClient';
+import ConfirmationDialog, {
+  ConfirmationDialogRef,
+} from '../../shared/components/ConfirmationDialog';
+import { Icon } from '../../shared/components/Icon';
+import View from '../../shared/components/View';
+import ViewTitle from '../../shared/components/ViewTitle';
+import Button from '../../shared/components/actions/Button';
+import Form from '../../shared/components/form/Form';
+import FormLabel from '../../shared/components/form/FormLabel';
+import InputError from '../../shared/components/form/InputError';
+import SelectInput from '../../shared/components/form/SelectInput';
+import SubmittingIndicator from '../../shared/components/form/SubmittingIndicator';
+import { useFlash } from '../../shared/hooks/flash';
+import useAuth from '../../shared/hooks/useAuth';
+
+export async function importLanguageGlossesLoader({
+  params,
+}: LoaderFunctionArgs) {
+  const language = await apiClient.languages.findByCode(
+    params.code ?? 'unknown'
+  );
+  const importLanguages = await apiClient.import.getLanguages();
+  return { language, importLanguages };
+}
+
+interface FormData {
+  import: string;
+}
+
+export default function ImportLanguageGlossesView() {
+  useAuth({ requireRole: [SystemRole.Admin] });
+  const { language, importLanguages } = useLoaderData() as {
+    language: GetLanguageResponseBody;
+    importLanguages: GetLanguageImportOptionsResponseBody;
+  };
+  const flash = useFlash();
+  const navigate = useNavigate();
+  const { t } = useTranslation();
+  const confirmationDialog = useRef<ConfirmationDialogRef>(null);
+  // This tracks if the backend is working on importing the glosses.
+  const [isImporting, setIsImporting] = useState<boolean>(false);
```

Comment:
We shouldn't need this import state anymore because the form will remain in loading state while the modal is open and the api request is in progress.

Reply:
I was using it to disable the `SelectInput` while the import was loading, but it looks like the user can't submit a new choice anyways so I'll remove it.

Reply:
Done